src/hex.rs

```diff
@@ -219,6 +219,51 @@ impl ToHex for [u8] {
     }
 }
 
+/// A struct implementing [`io::Write`] that converts what's written into it
+/// in an hex String
+///
+/// If you already have the data to be converted in a `Vec<u8>` use [`ToHex`]
+/// but if you have an encodable object, by using this you avoid the
+/// serialization to `Vec<u8>` by going directly to `String`.
+///
+/// Note that to achieve better perfomance than [`ToHex`] the struct must be
+/// created with the right `capacity` of the end hex string result so that the
+/// inner `String` doesn't re-allocate
+#[cfg(any(test, feature = "std", feature = "alloc"))]
```

Comment:
Some minor grammar fixes if you want them
```
/// A struct implementing [`io::Write`] that converts what's written to it into
/// a hex string.
///
/// If you already have the data to be converted in a `Vec<u8>` use [`ToHex`]
/// but if you have an encodable object, by using this you avoid the
/// serialization to `Vec<u8>` by going directly to `String`.
///
/// Note that to achieve better perfomance than [`ToHex`] the struct must be
/// created with the `capacity` of the final hex string so that the inner
/// `String` doesn't re-allocate.
```

TIL, for words starting with 'h' we use 'an' before them if the 'h' is silent (e.g., an hour) and we use 'a' before them if the 'h' is pronounced (e.g. a hat).

Reply:
thanks, will apply